retz-server/src/main/java/io/github/retz/scheduler/RetzScheduler.java

```diff
@@ -226,16 +226,28 @@ public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
                 offerStock.clear();
             }
 
-            ResourceQuantity total = new ResourceQuantity();
-            for (Protos.Offer offer : available) {
-                LOG.debug("offer: {}", offer);
-                Resource resource = ResourceConstructor.decode(offer.getResourcesList());
-                total.add(resource.toQuantity());
+            List<Job> jobs = new ArrayList<>();
+            switch (conf.getServerConfig().getJobQueueStrategy()) {
+                case Fit:
+                    ResourceQuantity total = new ResourceQuantity();
+                    for (Protos.Offer offer : available) {
+                        LOG.debug("offer: {}", offer);
+                        Resource resource = ResourceConstructor.decode(offer.getResourcesList());
+                        total.add(resource.toQuantity());
+                    }
+                    total.setNodes(offers.size());
+                    // TODO: change findFit to consider not only CPU and Memory, but GPUs and Ports
+                    jobs = JobQueue.findFit(planner.orderBy(), total);
+                    LOG.debug("found {} jobs fit for {}", jobs.size(), total.toString());
+                    break;
+                case All:
+                    jobs = JobQueue.findAll(planner.orderBy());
+                    LOG.debug("found {} jobs", jobs.size());
+                    break;
+                default:
+                    LOG.debug("skip unknown strategy");
```

Comment:
Considering startup check at `ServerConfiguration` constructor this must not happen; `AssertionError` is usually used in this software.